ethereum/blockcreation/src/test/java/tech/pegasys/pantheon/ethereum/blockcreation/EthHashBlockCreatorTest.java

```diff
@@ -136,4 +143,62 @@ public void createMainnetBlock1_fixedDifficulty1() throws IOException {
     // If we weren't setting difficulty to 2^256-1 a difficulty of 1 would have caused a
     // IllegalArgumentException at the previous line, as 2^256 is 33 bytes.
   }
+
+  @Test
+  public void createMainnetBlock1_noReward() {
+    final ExecutionContextTestFixture executionContextTestFixture =
+        ExecutionContextTestFixture.builder()
+            .protocolSchedule(
+                new ProtocolScheduleBuilder<>(
+                        GenesisConfigFile.fromConfig(
+                                "{\"config\": {\"ethash\": {\"fixeddifficulty\":1}}}")
+                            .getConfigOptions(),
+                        BigInteger.valueOf(42),
+                        Function.identity(),
+                        PrivacyParameters.DEFAULT,
+                        false)
+                    .createProtocolSchedule())
+            .build();
+
+    final EthHashSolver solver = new EthHashSolver(Lists.newArrayList(BLOCK_1_NONCE), new Light());
+
+    final PendingTransactions pendingTransactions =
+        new PendingTransactions(
+            TransactionPoolConfiguration.DEFAULT_TX_RETENTION_HOURS,
+            1,
+            TestClock.fixed(),
+            metricsSystem);
+
+    final EthHashBlockCreator blockCreator =
+        new EthHashBlockCreator(
+            BLOCK_1_COINBASE,
+            parent -> BLOCK_1_EXTRA_DATA,
+            pendingTransactions,
+            executionContextTestFixture.getProtocolContext(),
+            executionContextTestFixture.getProtocolSchedule(),
+            gasLimit -> gasLimit,
+            solver,
+            Wei.ZERO,
+            executionContextTestFixture.getBlockchain().getChainHeadHeader());
+
+    final MutableWorldState mutableWorldState =
+        executionContextTestFixture.getStateArchive().getMutable();
+    assertThat(mutableWorldState.get(BLOCK_1_COINBASE)).isNull();
+
+    final ProcessableBlockHeader header =
+        BlockHeaderBuilder.create()
+            .parentHash(Hash.ZERO)
+            .coinbase(BLOCK_1_COINBASE)
+            .difficulty(UInt256.ONE)
+            .number(1)
+            .gasLimit(1)
+            .timestamp(1)
+            .buildProcessableBlockHeader();
+
+    blockCreator.rewardBeneficiary(
+        mutableWorldState, header, Collections.emptyList(), Wei.ZERO, false);
```

Comment:
Probably worth creating another version of this test where `skipZeroBlockRewards` is `true`

Reply:
done